### PR TITLE
Added support for parallelized image plugins.

### PR DESCRIPTION
**Pull request by Chris210634**
@danmcp @rudolphpienaar @jbernal0019 

Added support in pman for specifying the **memory limit**, **cpu limit**, and **number of instances** for openshift jobs only. (If not running in openshift, these values will be ignored).

If these values are not provided by pfcon, then default values are used.

The following request to pman will run 2 instances of pl-simpledsapp:

```
pfurl --verb POST --raw --http pman-myproject.127.0.0.1.nip.io/api/v1/cmd --jsonwrapper 'payload' --msg \
        "{   \"action\": \"run\",
            \"meta\":  {
                \"cmd\": \"\$execshell \$selfpath/\$selfexec --prefix test- --sleepLength 0 /share/incoming /share/outgoing\",
                \"auid\": \"rudolphpienaar\",
                \"jid\": \"003\",
                \"threaded\": true,
                \"container\": {
                        \"target\": {
                            \"image\": \"fnndsc/pl-simpledsapp\",
                            \"cmdParse\": true
                        }
                },
                \"cpu_limit\": \"2000m\",
                \"memory_limit\": \"2Gi\",
                \"number_of_workers\": \"2\"
            }
        }" --quiet --jsonpprintindent 4
```

### CODE REVIEW

File: pman/openshiftmgr.py [hunk 1]
```diff
@@ -63,12 +64,12 @@ def schedule(self, image, command, name):
...
                                     "limits": {
-                                        "memory": "1024Mi",
-                                        "cpu": "2000m"
+                                        "memory": memory_limit,
+                                        "cpu": cpu_limit
                                     },
                                     "requests": {
-                                        "memory": "128Mi",
-                                        "cpu": "250m"
+                                        "memory": memory_limit,
```
Comment (danmcp) [inline pman/openshiftmgr.py]: Let's not change these values.

File: pman/openshiftmgr.py [hunk 2]
```diff
@@ -36,7 +36,8 @@ def init_openshift_client(self):
         self.kube_v1_batch_client = k_client.BatchV1Api()
 
     # The kubecfg secret being mounted in the third container (publish) is the one that is generated by the service account
-    def schedule(self, image, command, name):
+    def schedule(self, image, command, name, number_of_workers='1',
```
Comment (danmcp) [inline pman/openshiftmgr.py]: Why set defaults here and in the code that calls openshiftmgr?

File: pman/pman.py [hunk 3]
```diff
@@ -1814,7 +1827,8 @@ def t_run_process_openshift(self, *args, **kwargs):
             #
             if d_target['cmdParse']:
                 cmdparse_pod_name = self.jid + '-cmdparse'
-                self.get_openshift_manager().create_pod(str_targetImage, cmdparse_pod_name)
```
Comment (danmcp) [inline pman/pman.py]: This is the wrong line to change.  You want to change the schedule line below.

**Comment by rudolphpienaar** [comment]
@danmcp  just noting that I've looked this over and this PR seems your call. Note the e2e tests don't hit this case since they obviously use swarm.

**Comment by Chris210634** [comment]
@danmcp Updated with your changes.

File: pman/pman.py [hunk 4]
```diff
@@ -1801,6 +1801,19 @@ def t_run_process_openshift(self, *args, **kwargs):
             self.auid           = d_meta['auid']
             str_cmd             = d_meta['cmd']
 
+            if 'number_of_workers' in d_meta:
+                number_of_workers = d_meta['number_of_workers']
```
Comment (danmcp) [inline pman/pman.py]: Why is this not passed as an int to start with?

File: pman/openshiftmgr.py [hunk 5]
```diff
@@ -47,8 +48,8 @@ def schedule(self, image, command, name):
                 "name": name
             },
             "spec": {
-                "parallelism": 1,
-                "completions": 1,
+                "parallelism": int(number_of_workers),
```
Comment (danmcp) [inline pman/openshiftmgr.py]: I think this should be set to 1 or unset.

**Pull request closed without merging**